apps/web/app/(app)/simple/SimpleList.tsx

```diff
@@ -65,11 +66,25 @@ export function SimpleList(props: {
     .filter((m) => !toHandleLater[m.id])
     .map((m) => m.threadId);
 
+  const handleUnsubscribe = (id) => {
+    const message = props.messages.find((item) => item.id === id);
+    if (message) {
+      setUnsubscribed((currentUnsubscribed) => [
+        ...currentUnsubscribed,
+        message,
+      ]);
+    }
+  };
```

Comment:
Seems overly complicated. Why not just add the id to the array? Why the need for find?

Also, you could make it a Set or object instead of an array of ids. A bit more efficient and then wouldn't need to do `.find`.